octree/include/pcl/octree/impl/octree_iterator.hpp

```diff
@@ -281,6 +281,79 @@ namespace pcl
 
       return (*this);
     }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::OctreeFixedDepthIterator (unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (depth_arg), depth_(depth_arg)
+    {
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::OctreeFixedDepthIterator (OctreeT* octree_arg, unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (octree_arg, depth_arg), depth_(depth_arg)
+    {
+      this->reset (depth_arg);
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::OctreeFixedDepthIterator (const OctreeFixedDepthIterator& iterator)
+    {
+      // initialize with the input iterator
+      *this = iterator;
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::~OctreeFixedDepthIterator ()
+    {
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    void OctreeFixedDepthIterator<OctreeT>::reset ()
+    {
+      OctreeBreadthFirstIterator<OctreeT>::reset ();
+
+      // Set the current_state_ to the right depth
+      while (this->current_state_ && (this->getCurrentOctreeDepth () != depth_))
+      {
+        ++*this;
+      }
+    }
```

Comment:
This method is very inefficient. You should apply a depth first search approach to reach your desired initialization state.